### ISSUE TITLE
Breaking change in separation_from()?

### ISSUE THREAD

**Issue by lucabaldini** [skyfielders/python-skyfield#414]
Hi Brandon---abusing of your patience again.

In skyfield 1.20 I used to be able to do `pos.separation_from(target_pos)`, where `pos` is holding an array of position at different times and `target_pos` is a fixed (scalar) position. Apparently this isn't possible anymore, and I get a ValueError from `angle_between()` because the two operands cannot be broadcast together.

Glancing through the commits, I believe this is the relevant one
https://github.com/skyfielders/python-skyfield/commit/edd79543e5289de5eae65c307ef550d5bd16ad72
i.e., you got rid of the magic that made the broadcast work.

This is not necessarily a problem in skyfield, and I think I can find my way around it, but I have two questions:

1. is the new behavior intended?
2. do you have specific suggestions about how I would support my code against different skyfield versions? 

Thanks in advance and keep up with your amazing work!
Luca

**Comment by brandon-rhodes** [comment]
> 1. is the new behavior intended?

No, my intention was not to deliberately break a use case — but as there were, alas, no tests for `separation_from()` in the case of two angles with different NumPy shapes, I did not know about the change in behavior.

> 2. do you have specific suggestions about how I would support my code against different skyfield versions?

No. In general, working code against anything — including Python itself — can only be maintained by specifying exact versions of what you depend on. So depending on a specific version of Skyfield is recommended. Skyfield itself, it turns out, can be sensitive to things like the NumPy version someone has installed, and it has been tricky to make sure I only use NumPy features that work against all NumPy versions someone might be using!

> Thanks in advance and keep up with your amazing work!

Thanks for the encouragement!

I'll see about getting `separation_from()` fixed and getting a test case written.

The topic, by the way, of vectors of different broadcast dimensions is a big one that I'm hoping to tackle soon, so that folks can do things like have *n* dates and *m* comets and get all combinations computed at once!

**Comment by lucabaldini** [comment]
Thanks---makes perfect sense.

Luca

**Comment by brandon-rhodes** [comment]
There! I think I've solved it by adding a bit more logic. Could you try:
```
pip install https://github.com/skyfielders/python-skyfield/archive/master.zip
```
— and see if the issue is now fixed for you?

**Comment by brandon-rhodes** [comment]
I have just released Skyfield 1.26, which includes this fix. Thanks again for letting me know!
